vortex-roaring/src/integer/mod.rs

```diff
@@ -1,34 +1,28 @@
-use std::sync::{Arc, RwLock};
-
 use compress::roaring_encode;
-use croaring::{Bitmap, Native};
-use vortex::array::{Array, ArrayKind, ArrayRef};
-use vortex::compress::EncodingCompression;
-use vortex::compute::ArrayCompute;
-use vortex::encoding::{Encoding, EncodingId, EncodingRef};
-use vortex::formatter::{ArrayDisplay, ArrayFormatter};
+use croaring::{Bitmap, Portable};
+use serde::{Deserialize, Serialize};
+use vortex::array::primitive::{Primitive, PrimitiveArray};
+use vortex::buffer::Buffer;
 use vortex::ptype::PType;
-use vortex::serde::{ArraySerde, EncodingSerde};
-use vortex::stats::{Stats, StatsSet};
-use vortex::validity::ArrayValidity;
-use vortex::validity::Validity;
-use vortex::{impl_array, ArrayWalker};
+use vortex::stats::ArrayStatisticsCompute;
+use vortex::validity::{ArrayValidity, LogicalValidity};
+use vortex::visitor::{AcceptArrayVisitor, ArrayVisitor};
+use vortex::{impl_encoding, ArrayFlatten, OwnedArray};
 use vortex_error::{vortex_bail, vortex_err, VortexResult};
-use vortex_schema::DType;
+use vortex_schema::Nullability::NonNullable;
 
 mod compress;
 mod compute;
-mod serde;
-mod stats;
 
-#[derive(Debug, Clone)]
-pub struct RoaringIntArray {
-    bitmap: Bitmap,
+impl_encoding!("vortex.roaring_int", RoaringInt);
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct RoaringIntMetadata {
     ptype: PType,
-    stats: Arc<RwLock<StatsSet>>,
+    length: usize,
```

Comment:
Can you leave a comment that this is here because we want fast access to length that doesn't involve deserialising the roaring bitmap.

@robert3005 any ideas on how to avoid the deserialization overhead?  Should we shove the bitmap into metadata instead of into a buffer?

Reply:
I don’t think so, maybe depends on the size? We should keep it like this and see if we ever have an issue. Agree it could be bad if somewhere forgets to change the length in metadata :/